### PR TITLE
omit password in string method

### LINKED ISSUE #139: Do not log password config.
## Bug Report

Please answer these questions before submitting your issue. Thanks!

1. What did you do? If possible, provide a recipe for reproducing the error.
create dumpling in DM
2. What did you expect to see?
no password in log file
3. What did you see instead?
password in log file
4. Versions of the cluster

    - Dumpling version (run `dumpling -V`):

        ```
        (paste Dumpling version here)
        ```

    - Source database version (execute `SELECT version();` in a MySQL client):

        ```
        (paste source database version here)
        ```

    - Other interesting information (system version, hardware config, etc):

        >
        >

5. Operation logs
   - Please upload `dumpling.log` for Dumpling if possible
   - Other interesting logs
   
6. Configuration of the cluster and the task
   - running command for Dumpling if possible

### PR DESCRIPTION
**Pull request by GMHDBJD**
### What problem does this PR solve? 
close #139 

### What is changed and how it works?
- add json string method, omit the password

### Check List 

Tests 

 - Manual test (add detailed scripts or steps below)
    manually check with DM

### CODE REVIEW

**Comment by GMHDBJD** [comment]
@lichunzhu @csuzhangxc PTAL

File: v4/export/config.go [hunk 1]
```diff
@@ -14,51 +15,51 @@ import (
...
-	Databases []string
-	Host      string
-	User      string
-	Port      int
-	Password  string
+	Databases []string `json:"Databases"`
+	Host      string   `json:"Host"`
+	User      string   `json:"User"`
+	Port      int      `json:"Port"`
+	Password  string   `json:"-"` // omit it for privacy
```
Comment (kennytm) [inline v4/export/config.go:22]: only this line is necessary. all other lines are redundant.

File: v4/export/config.go [hunk 2]
```diff
@@ -14,51 +15,51 @@ import (
...
-	Threads int
+	Threads int `json:"Threads"`
 
-	LogLevel  string
-	LogFile   string
-	LogFormat string
+	LogLevel  string `json:"LogLevel"`
+	LogFile   string `json:"LogFile"`
+	LogFormat string `json:"LogFormat"`
 	Logger    *zap.Logger
```
Comment (kennytm) [inline v4/export/config.go:34]: i guess you need to skip this field too.

File: v4/export/config.go (hunk 1 above)
Comment (GMHDBJD) [reply v4/export/config.go]: address in cc939a3

File: v4/export/config.go [hunk 3]
```diff
@@ -15,51 +15,51 @@ import (
...
+	CsvDelimiter  string
+
+	TableFilter        filter.Filter
+	Rows               uint64
+	Where              string
+	FileType           string
+	CompleteInsert     bool
+	EscapeBackslash    bool
+	DumpEmptyDatabase  bool
+	OutputFileTemplate *template.Template
```
Comment (kennytm) [inline v4/export/config.go]: oh the `TableFilter` and `OutputFileTemplate` also cannot be encoded into JSON

Comment (GMHDBJD) [reply v4/export/config.go]: address in 4f7f84a

**Pull request merged**